src/Illuminate/Http/Resources/ConditionallyLoadsAttributes.php

```diff
@@ -66,22 +67,29 @@ protected function mergeData($data, $index, $merge, $numericKeys)
      * Remove the missing values from the filtered data.
      *
      * @param  array  $data
-     * @param  bool  $numericKeys
      * @return array
      */
-    protected function removeMissingValues($data, $numericKeys = false)
+    protected function removeMissingValues($data)
     {
+        $numericKeys = true;
+
         foreach ($data as $key => $value) {
             if (($value instanceof PotentiallyMissing && $value->isMissing()) ||
                 ($value instanceof self &&
                 $value->resource instanceof PotentiallyMissing &&
                 $value->isMissing())) {
                 unset($data[$key]);
             }
+            $numericKeys = $numericKeys && is_numeric($key);
```

Comment:
I think we should determine `$numericKeys` *after* the missing values are removed. Otherwise something like `[1 => 'foo', 'string' => new MissingValue]` returns `{1: "foo"}` instead of `["foo"]`.

Reply:
I really thought I had it inside an else clause. Maybe I am too used with using guard clauses in my code (I would have used a `continue` inside the `if`). Thanks for the heads up. I added a test to check for this case.

As of using the one-liner after the `while`, my thinking was to use it inside the loop as it was already looping over all the elements anyway, so I thought a simple operation like that (boolean addition) would'n hurt performance as much as doing another run on the elements. Also I thought the intention of the code is more clear.

If you think your suggestion would perform better, or communicate better, please let me know and I update it.

Reply:
Your approach is definitely faster, I just wanted to provide a shorter alternative.